Mage.Sets/src/mage/cards/m/MangarasEquity.java

```diff
@@ -0,0 +1,161 @@
+        staticText = "choose black or red";
+    }
+
+    private MangarasEquityColorChoiceEffect(final MangarasEquityColorChoiceEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Permanent permanent = game.getPermanentEntering(source.getSourceId());
+        if (permanent == null) {
+            return false;
+        }
+
+        ChoiceColor choice = new ChoiceColor(true, "Choose a color.", permanent);
+        choice.removeColorFromChoices("White");
+        choice.removeColorFromChoices("Blue");
+        choice.removeColorFromChoices("Green");
+
+        if (controller == null || !controller.choose(outcome, choice, game)) {
+            return false;
+        }
+
+        game.informPlayers(permanent.getLogName() + ": " + controller.getLogName() + " has chosen " + choice.getChoice());
+        game.getState().setValue(permanent.getId() + "_color", choice.getColor());
+        permanent.addInfo("chosen color", CardUtil.addToolTipMarkTags("Chosen color: " + choice.getChoice()), game);
+
+        return true;
+    }
+
+    @Override
+    public Effect copy() {
+        return new MangarasEquityColorChoiceEffect(this);
+    }
+}
+
+class MangarasEquityTriggeredAbility extends TriggeredAbilityImpl {
+
+    MangarasEquityTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new DamageTargetEffect(0));
+    }
+
+    private MangarasEquityTriggeredAbility(final MangarasEquityTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DAMAGED_PLAYER
+                || event.getType() == GameEvent.EventType.DAMAGED_PERMANENT;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        Permanent mangarasEquity = game.getPermanentOrLKIBattlefield(getSourceId());
+        if (mangarasEquity == null) {
+            return false;
+        }
+
+        if (event.getType() == GameEvent.EventType.DAMAGED_PLAYER) {
+            if (!event.getTargetId().equals(getControllerId())) {
+                return false;
+            }
+        } else {
+            Permanent damagedPermanent = game.getPermanent(event.getTargetId());
+            if (damagedPermanent == null || !damagedPermanent.isControlledBy(getControllerId())
+                    || !damagedPermanent.isCreature() || !damagedPermanent.getColor().shares(ObjectColor.WHITE)) {
+                return false;
+            }
+        }
+
+        Permanent damageSource = game.getPermanent(event.getSourceId());
+        ObjectColor chosenColor = (ObjectColor) game.getState().getValue(mangarasEquity.getId() + "_color");
+        if (damageSource != null && damageSource.isCreature() && damageSource.getColor().shares(chosenColor)) {
+            for (Effect effect : getEffects()) {
```

Comment:
why not store `this.getEffects().setValue("damage", damageAmount);` and use `SavedDamageValue` in the constructor of the `DamageTargetEffect`?

Reply:
Thanks for the tip!